simon8491/main/java/pl/coderslab/book_api/controller/BookController.java

```diff
@@ -0,0 +1,47 @@
+package pl.coderslab.book_api.controller;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+import pl.coderslab.book_api.model.Book;
+import pl.coderslab.book_api.service.BookService;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/books")
+public class BookController {
+    private final BookService bookService;
+
+    public BookController(BookService bookService) {
+        this.bookService = bookService;
+    }
+
+    @GetMapping("")
+    public List<Book> getAllBooks() {
+        return bookService.getAllBooks();
+    }
+    @PostMapping("")
+    public Book addBook(@RequestBody Book book) {
+        return bookService.addBook(book);
+    }
+    @GetMapping("{id}")
+    public Book getBook(@PathVariable Long id) {
+        return this.bookService.getBook(id).orElseThrow(() ->
+                new ResponseStatusException(HttpStatus.NOT_FOUND, "Book with id: " + id + " not found"));
+    }
+    @PutMapping("")
+    public ResponseEntity<Book> updateBook(@RequestBody Book book) {
+        var updatedBook = bookService.updateBook(book);
```

Comment:
CLEAN-CODE: Kde je mozne je dobre pouzivat final nemenne promenne.

```suggestion
final var updatedBook = bookService.updateBook(book);
```

Reply:
Opraveno